modules/lib/core/index.d.ts

```diff
@@ -52,23 +67,115 @@ export interface PublishInfo {
     first?: string;
 }
 
-export interface Component<Config extends object = object, Regions extends Record<string, Region> = Record<string, Region>> {
-    config: Config;
-    descriptor: string;
+export interface FragmentComponent {
+    type: 'fragment'
+    fragment: string;
     path: string;
-    type: 'page' | 'layout' | 'part';
+}
+
+export interface LayoutComponent<
+    Descriptor extends ComponentDescriptor = ComponentDescriptor,
+    Config extends NestedRecord = NestedRecord,
+    Regions extends
+        Record<string, Region<(FragmentComponent | PartComponent | TextComponent)[]>> = 
+        Record<string, Region<(FragmentComponent | Part          | TextComponent)[]>>
```

Comment:
I think you should (for all Component-types) reorder the type parameters so that `Config` comes first. In many cases `Config` is the only thing you need, so it's nice to only have to pass in `Config`. I suggest `Regions` still come last to make this as similar as possible to `PartComponent`
 1. Config
 2. Descriptor
 3. Regions

Reply:
Well, if we have a registry where descriptor is the key, the descriptor should be all you need, not config?
With only descriptor the "full" shape is described (via registry).
With only config, descriptor remains `${string}:${string}`

Reply:
With descriptor first, this is now possible:

```typescript
getComponent<Component<'app.name:partDescriptor'>>()
```

Fragment and text doesn't have descriptors, but it doesn't matter because they don't have Config or Regions either.

```typescript
getComponent<FragmentComponent>()
```